### DISCUSSION TITLE
How to do a figurine notation?

### DISCUSSION THREAD

**Discussion by mliebelt** [mliebelt/pgn-viewer#249] [Ideas]
See the issue #248 about the context. I think a discussion may help support the topic. Here are my thought to it:

* What approaches are available?
  * Use a different font, where the characters R,N,B,K,Q,P are replaced by a figurine symbol. Approach taken by Lichess/Chesstempo.
  * Use real icons / images / SVG, and replace the character in PGN by that symbol. This could be done by CSS only then (and having the figure be more rich like `Nf3` ==> `<span class="figure knight white">N</span>f3`
* What will possibly break when doing this?
  * Internationalization: Figure characters are translated, and it may be more difficult to do that then. Or we have to ignore translations for figures in case of figurine notation.

**Comment by mliebelt** [comment]
So collected some more information, and found locally some more font files for chess. So here is something like a plan:

1. Ensure that the figure is made explicit in the notation, without any other change, so that the figure may be "styled" by CSS. (was not necessary, if the right font is used)
2. Try out the styling, to see if that works well.
3. Try out if setting the correct font for the figure is sufficient.
4. Then try out different languages, and see that the figurine notation then breaks.
5. Ensure that the locale information is **not** used when figurine is set.

Some resources I have used:

* [Using Web Fonts](https://css-tricks.com/understanding-web-fonts-getting/) for doing the CSS stuff.
* [Font Converter](https://cloudconvert.com/woff-to-ttf) if woff was not available.
* [ChessTempo Web Font](chessalphanew-webfont.vers1.woff)
* [Lichess Web Font](https://lichess1.org/assets/_KLITAX/font/lichess.chess.woff2)
* [FontForge](https://fontforge.org/en-US/) Used that tool to reduce fonts to only the chess symbols. Works like a charm, allows to create at least WOFF directly.

**Comment by mliebelt** [comment]
I found as tool FontForge, and was able to minimize all fonts to only the 6 pieces needed. So the foot print is not too much, and perhaps we can allow to have 4-6 figurine fonts used. Next step should be how to have conditional builds with different sets of resources: everything, no figurine, only a subset of piece styles, a subset of i18n locales, ...

**Reply by braoult** [reply]
I wonder why Lichess did prefer to have a 28 characters in their font (in fact all characters needed for full move notation). Any idea ?

**Reply by mliebelt** [reply]
I think the reason is, because they can use the font for everything (including a-h, 1-8 and some more characters like -,x,+ and #. If we try to split the move into figure, san and nags, then this problem will go away. I think that is the reason.

**Reply by braoult** [reply]
Thanks for `fontforge` hint I just installed it (and `birdfont`) and will have a look.

**Reply by braoult** [reply]
> I think the reason is, because they can use the font for everything (including a-h, 1-8 and some more characters like -,x,+ and #. If we try to split the move into figure, san and nags, then this problem will go away. I think that is the reason.

Yes, this is what I think, this allows them to have coherent moves display on  any device, independently from available fonts. I wonder if we should consider this approach, even extending it with NAGs which could maybe not exist on client-side font ?

**Comment by mliebelt** [comment]
There were now figurine notations possible, support for some 5 fonts or so, so this question should be answered. See #248 for some more information.
